### ISSUE TITLE
Swapping components in layout through button callback no longer works with threading enabled

### ISSUE THREAD

**Issue by TheoMathurin** [holoviz/panel#6671]
Panel 1.4.0
Bokeh 3.4.0

When threading is enabled in Panel 1.4.0, it is not possible to change the order of components in-place within a container (e.g. a column) through a button callback. It worked in Panel 1.3.8.

```python
import panel as pn


def swap_buttons(event):
    buttons[0], buttons[1] = buttons[1], buttons[0]


button1 = pn.widgets.Button(name='Button 1', button_type='primary')
button2 = pn.widgets.Button(name='Button 2', button_type='primary')
buttons = pn.Column(button1, button2)
switch = pn.widgets.Button(name='Switch buttons', button_type='success', on_click=swap_buttons)

pn.Row(buttons, switch).servable()
```

This will work with `panel serve app.py` but not with `panel serve app.py --num-threads 2`. There is no error thrown server-side or in the browser.

**Comment by TheoMathurin** [comment]
Thanks for investigating this.

Should I write a pure Bokeh MRE and file the issue there then?

**Comment by philippjfr** [comment]
Done: https://github.com/bokeh/bokeh/issues/13806

**Comment by philippjfr** [comment]
The only reason it works in non-threaded mode in Panel is because you did this:

```python
buttons[1], buttons[0] = buttons[0], buttons[1]
```

The correct way to write this reproduces this issue reliably even when non-threaded:

```python
    buttons[:] = buttons[::-1]
```

**Comment by TheoMathurin** [comment]
Alright thanks, looking at your issue which does not mention threading, I was wondering why it worked with threading disabled.

**Comment by TheoMathurin** [comment]
This issue is associated with the v1.4.2 milestone. But I think we can close it as the fix was included in Bokeh 3.4.1 which works with v1.4.1.

**Comment by philippjfr** [comment]
Yep, thanks!
